File: Code/Visitor.dotNet/Visitor.dotNet/Visitor.dotNet/Models/VisitableSquare.cs
using Visitor.Visitors;

namespace Visitor.Models
{
    /// <summary>
    /// This class extends a square so that it can be visited
    /// </summary>
    internal class VisitableSquare : Square, IVisitableObject
    {
        public VisitableSquare(double side) : base(side)
        {
        }

        public void Accept(IVisitor v)
        {
            v.Visit(this);
        }
    }
}

File: Code/Visitor.Java/src/models/VisitableSquare.java
package models;

import visitors.IVisitor;

/**
 * This class extends a square so that it can be visited
 */
public class VisitableSquare extends Square implements IVisitableObject {

	public VisitableSquare(double side) {
		super(side);
	}

	@Override
	public void accept(IVisitor v) {
		v.visit(this);		
	}

}
